source/cipher.c

```diff
@@ -88,6 +88,32 @@ static enum aws_cryptosdk_sha_version aws_cryptosdk_which_sha(enum aws_cryptosdk
     }
 }
 
+static bool alg_properties_equal(
+    const struct aws_cryptosdk_alg_properties alg_props1, const struct aws_cryptosdk_alg_properties alg_props2) {
```

Comment:
alg_properties structures should always be handled by reference, and should only point to the instances defined in cipher.h. It is expected that you should be able to confirm that two alg_props are equivalent by pointer equality.

Reply:
This function was added in PR #378 and should be handled there. @angelhof can you take a look at these comments?

Reply:
I addressed this in https://github.com/danielsn/aws-encryption-sdk-c/pull/2.